src/sagas/repository.js

```diff
@@ -88,13 +89,35 @@ function* pollJobWorker(jobId, jobTypeSuccess, jobTypeFailure) {
  * Datasets.
  */
 
-export function* createDataset({ payload }) {
+export function* createDataset() {
+  const dataset = yield select(getCreateDataset);
+  const payload = {
+    name: dataset.name,
+    description: dataset.description,
+    contents: [
+      {
+        source: {
+          studyName: dataset.study,
+          assetName: dataset.asset,
+        },
+        rootValues: dataset.ids,
+      },
+    ],
+  };
+  console.log(payload);
   try {
     const response = yield call(authPost, '/api/repository/v1/datasets', payload);
     const jobId = response.data.id;
     yield put({
       type: ActionTypes.CREATE_DATASET_JOB,
-      payload: { data: response, createdDataset: payload },
+      payload: { data: response, createdDataset: { jobId: payload } },
```

Comment:
You can't use a dynamic value like `jobId` as the key in the object here. It'd be nice to pull it out into the top level:
```javascript
{
    data: response,
    jobId: jobId,
    createdDataset: datasetRequest
}
```